bin/dev_env_menu

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# This script provides a menu of Service Broker tests to be run.
+# These tests are intended to be run within the Service Broker test
+# container that is created as part of the development environment
+# (via the 'start' script).
+
+# Run this from the root project directory so that rspec and Cucumber
+# tests can be discovered.
+cd "$(dirname "$0")/.." || (echo "Could not cd to parent dir"; exit 1)
+
+function main_menu() {
+
+  local options=("Run rspec unit tests" \
+           "Run integration (non-E2E) Cucumber tests" \
+           "Select from a list of Cucumber features to test" \
+           "Select from a list of Cucumber scenarios to test" \
+           "Run a bash shell in test container" \
+           "Exit and clean up development environment" )
+
+  while true; do
+    PS3=$'\nPlease select tests to run: '
+    echo
+    select option in "${options[@]}"; do
+      case $REPLY in
+          1) unit_test; break ;;
+          2) integration_test; break ;;
+          3) feature_menu; break ;;
+          4) scenario_menu; break ;;
+          5) bash; break ;;
+          6) exit 0
+      esac
+    done
+  done
+}
+
+function feature_menu() {
+  PS3=$'\nPlease select a feature to test: '
+  sub_menu "Feature: "
+}
+
+function scenario_menu() {
+  PS3=$'\nPlease select a scenario to test: '
+  sub_menu "Scenario: "
+}
+  
+function sub_menu() {
+  # Search for instances of the specified pattern in Cucumber dry run output
+  pattern="$1"
+  IFS=$'\n'
+  local options=( $(dry_run | grep "$pattern" | sed "s/[ ]*$pattern[ ]*//") )
+  unset IFS
+  
+  # Add an option to return to the main menu
+  options+=("Return to main menu")
```

Comment:
(Nit) I noticed that return to main menu is listed last. Since the size of the list changes as we add scenarios and features, it might be better to make 'return to main menu' always be '0'.

Reply:
Unfortunately, the bash `select` feature doesn't allow indexing responses from `0`.
BUT, I will change the ordering so that `Return to main menu` is always option `1`.